openpgp/keys.go

```diff
@@ -615,6 +621,9 @@ func (e *Entity) serializePrivate(w io.Writer, config *packet.Config, reSign boo
 			return
 		}
 		if reSign {
+			if ident.SelfSignature == nil {
+				return goerrors.New("openpgp: request to re-sign identity without valid self-signature")
```

Comment:
I was slightly unsure about this one, as I suppose you could go ahead and create a signature in this case, but if I understand correctly you'd have to pick a bunch of parameters (which are otherwise taken from the existing sig), so my guess is that would be out of scope for this function.

Reply:
Myeah, I agree this seems reasonable, thanks! I'll just slightly re-word the error:
```suggestion
				return goerrors.New("openpgp: can't re-sign identity without valid self-signature")
```